src/appleseed/renderer/modeling/entity/entity.h

```diff
@@ -223,6 +246,26 @@ void invoke_update_asset_paths(
         entity.update_asset_paths(mappings);
 }
 
+template <typename EntityCollection>
+bool invoke_on_render_begin(
+    EntityCollection&                   entities,
+    const Project&                      project,
+    const BaseGroup*                    parent,
+    OnRenderBeginRecorder&              recorder,
+    foundation::IAbortSwitch*           abort_switch)
+{
+    for (auto& entity : entities)
+    {
+        if (foundation::is_aborted(abort_switch))
+            return false;
+
+        if (!entity.on_render_begin(project, parent, recorder, abort_switch))
+            return false;
```

Comment:
Is that better ?

```cpp
if (foundation::is_aborted(abort_switch) ||
    !entity.on_render_begin(project, parent, recorder, abort_switch)
    return false;
```
Just one branch instruction, less cpu prediction

Reply:
I'm pretty sure the compiler will compile `||` as a series of branches, so no gains there I'm afraid... (Again, this is done before rendering so not as performance-critical as code that happens during rendering.)

Reply:
Ok, so it's nano-optimization...
Not important